### ISSUE TITLE
Redundant JARs in the published maven artifact.

### ISSUE THREAD

**Issue by metaphore** [tinify/tinify-java#9]
`gson` and `okhttp` are transitive dependencies and get resolved properly for the client projects. However, it appears that the unnecessary JARs are bundled with the resulting artifact published on Maven Central.

I'm not very familiar with Maven configuration, but I guess that should be a quick fix. 

![image](https://user-images.githubusercontent.com/6159734/94989689-f99f7e00-05b9-11eb-9170-0edb82f97d84.png)

**Comment by eddyh** [comment]
Hi @metaphore,

`gson` and `okhttp` are direct dependencies of `tinify-java`, which is why they're included in the `pom.xml` and jar.

**Comment by metaphore** [comment]
They are for sure, but the fact they are declared as `compile` dependencies in `pom.xml` already let the local Maven client to resolve the dependencies from Maven Central repo.

A bit of background:
I use Gradle for my desktop Java app and have the lib included as a Maven dependency 
```
implementation "com.tinify:tinify:$tinifyVersion"
```
I can see it pulls `gson` and `okhttp` both from Maven Central. In the general case, there's no issue at all, they don't clash. But when I bundle a fat Jar of my app, those duplicating `gson` and `okhttp` jars are also included (next to their actual compiled classes) because they are part of the `tinify` classpath.

In other words, there's no need to include those jars in the distribution (at least not for the Maven Central). Or am I missing something?

**Comment by eddyh** [comment]
Thanks for the info. I've done some further investigating and found that the dependencies are included to make it valid OSGI bundle. As it's not causing any issues for you, I think we can leave it as is and close this issue. 

However, if there exists a best practice to improve this that we're not aware of, we'll be happy to know. You're also welcome to submit a PR if that's the case.

**Comment by metaphore** [comment]
No worries, it's not critical by any means and can easily be altered on the client-side if needed. I just wanted to bring attention to rather an atypical way the distribution is bundled for Maven Central. I'm not an expert in Maven and only familiar with the artifacts from the Gradle client perspective so definitely cannot debate what's right and what's not in relation to Maven jars.

I guess it's all good and totally fine to leave everything as is. @eddyh, thanks for checking on it anyway!
